runtime/protocol/http-client/common/src/aws/smithy/kotlin/runtime/http/engine/internal/HttpClientMetrics.kt

```diff
@@ -122,6 +100,41 @@ public class HttpClientMetrics(
         "The amount of time after a request has been sent spent waiting on a response from the remote server",
     )
 
+    private val asyncHandles = listOfNotNull(
+        meter.createAsyncUpDownCounter(
+            "smithy.client.http.connections.limit",
+            { it.record(_connectionsLimit.value) },
+            "{connection}",
+            "Max connections configured for the HTTP client",
+        ),
+        meter.createAsyncUpDownCounter(
+            "smithy.client.http.connections.usage",
+            ::recordConnectionState,
+            "{connection}",
+            "Current state of connections (idle, acquired)",
+        ),
+        meter.createAsyncUpDownCounter(
+            "smithy.client.http.requests.limit",
+            { it.record(_requestConcurrencyLimit.value) },
+            "{request}",
+            "Max concurrent requests configured for the HTTP client",
+        ),
+        meter.createAsyncUpDownCounter(
+            "smithy.client.http.requests.usage",
+            ::recordRequestsState,
+            "{request}",
+            "The current state of HTTP client request concurrency (queued, in-flight)",
+        ),
+        threadStateCallback?.let { callback ->
+            meter.createAsyncUpDownCounter(
+                "smithy.client.http.threads.count",
+                { measurement -> recordThreadState(measurement, callback) },
+                "{thread}",
+                "Current state of HTTP engine threads (active, running)",
```

Comment:
How are "running" threads different from "active" ones? I see only "total" and "active" are recorded.

Reply:
You're right, "active" and "running" are not the correct terms here. I can change those to "total" and "active".